src/Symfony/Component/HttpClient/Tests/HttpClientTestCase.php

```diff
@@ -466,4 +466,13 @@ public function testMisspelledScheme()
 
         $httpClient->request('GET', 'http:/localhost:8057/');
     }
+
+    public function testNoRedirectWithInvalidLocation()
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+
+        $response = $client->request('GET', 'http://localhost:8057/302-no-scheme');
+
+        $this->assertSame(302, $response->getStatusCode());
```

Comment:
The fact that no redirection happens is implicit because otherwise the status code wouldn't be 302 as the client would follow the location header URL?

Reply:
exactly: when a redirection target cannot be resolved, the client just stops there